core/parser.mly

```diff
@@ -426,6 +427,8 @@ fun_declarations:
 fun_declaration:
 | tlfunbinding                                                 { fun_binding ~ppos:$loc($1) None $1 }
 | signatures tlfunbinding                                      { fun_binding ~ppos:$loc($2) (fst $1) ~unsafe_sig:(snd $1) $2 }
+| match_tlfunbinding                                           { match_fun_binding ~ppos:$loc($1) None $1 }
+| signatures match_tlfunbinding                                { match_fun_binding ~ppos:$loc($2) (fst $1) ~unsafe_sig:(snd $1) $2 }
 
```

Comment:
It seems that `match` can be used in toplevel bindings or anonymous functions but not ordinary function bindings (or toplevel operator definitions).  Why not?  I think the desugaring should work for them too without problems.

On the other hand, I'm not a huge fan of the duplication of rules here and below.

Reply:
Added support to ordinary function bindings. Consider they are calling different sugar constructors, I don't know how to merge them.